### ISSUE TITLE
Delay acorn apps deployment waiting for secrets got created in a workload's ns

### ISSUE THREAD

**Issue by denist-huma** [acorn-io/acorn#583]
One has to delete app pods because they ran into connection errors, at least I saw this with Redis.
Unfortunately, these connection errors in "todo-app" containers don't exit the app.
It's not possible to delay Acorn apps start to wait for secret creation.
The [dependsOn](https://docs.acorn.io/reference/acornfile#dependson) works only for Acorn containers.
[Linking Acorn Apps](https://docs.acorn.io/running/linking-acorns) works for other running Acorn services only.
It's not possible to attach to the "ready" status of Redis CR.

I used acorn to run my example todo app that uses KubeDB's MongoDB, Redis, S3 bucket I create with my operators.
https://gist.github.com/denist-huma/aed27d49cb9f7449cf299824c30af0c9

This a work flow I discovered:

### Run a todo-app example as an Acorn app

To start your Acorn app just run:

`acorn run -n todo-app-acorn .`
 
 ### Create KubeDB's MongoDB, Redis, S3 bucket for app

First you need to get a generated namespace name for todo-app-acorn. [user configured namespace and service account for acorns](https://github.com/acorn-io/acorn/issues/341) is not supported in the moment.

> when acorn actually launches k8s workloads, it does so in a generated namespace per app.

```sh
$ k get ns | grep todo-app-acorn
todo-app-acorn-dd969d4b-605   Active   68s
```

Create a KubeDB's MongoDB, Redis, S3 bucket for app in this generated namespace.

```sh
$ k apply -f redis-claim.yaml -f bucket.yaml -f mongodb-user.yaml -n todo-app-acorn-dd969d4b-605
redisclaim.redis.operator.huma.com/todo-app-rc created
s3bucket.storage.huma.com/todo-app-bucket created
mongodbuser.db.operator.huma.com/todo-app-db created
```

- Wait till Acorn app ingress get assigned a temporary DNS.
- One has to delete app pods because they ran into connection errors. Unfortunately, these connection errors in my containers don't exit app.

**Comment by cjellick** [comment]
This is good feedback.

What do you mean by " external secrets"?

**Comment by denist-huma** [comment]
@cjellick I am not sure **anymore** that issue is on your side. I am still discovering what is going on. Had no time to come back to this app.
As I know from my practice, generally when deployment has a secretKeyRef it will not create a pod until secret exist.
I still wonder why my exmple app has error connecting to redis and why I have to restart it.

```yaml
apiVersion: apps/v1
kind: Deployment
spec:
  template:
    spec:
      automountServiceAccountToken: false
      containers:
      - command:
        - celery
        - -A
        - app
        - worker
        - --loglevel=INFO
        env:
        - name: REDIS_URL
          valueFrom:
            secretKeyRef:
              key: redis_url
              name: todo-app-rc-secret
 ```

**Comment by denist-huma** [comment]
Not an iisue anymore, solved after v0.14.0 acorn install
